lib/core/src/display/shape/text/text_field/keyboard.rs

```diff
@@ -0,0 +1,197 @@
+            if let Ok(key) = event.key().parse::<Key>() {
+                frp_key_released.event.emit(key);
+            }
+        });
+        binding.set_copy_handler(move |is_cut| {
+            if is_cut {
+                frp_cut.event.emit(())
+            } else {
+                frp_copy.event.emit(());
+            }
+            frp_text_to_copy.behavior.current_value()
+        });
+        binding.set_paste_handler(move |text_to_paste| {
+            frp_paste.event.emit(text_to_paste);
+        });
+        binding
+    }
+}
+
+
+// === Private ===
+
+impl TextFieldFrp {
+
+    fn copy_action_lambda(cut:bool, text_field_ptr:Weak<RefCell<TextFieldData>>)
+    -> impl Fn() -> String {
+        move || {
+            match text_field_ptr.upgrade() {
+                Some(text_field) => {
+                    let mut text_field_ref = text_field.borrow_mut();
+                    let result             = text_field_ref.get_selected_text();
+                    if cut { text_field_ref.edit(""); }
+                    result
+                },
+                None => default()
+            }
+        }
+    }
+
+    fn paste_action_lambda(text_field_ptr:Weak<RefCell<TextFieldData>>) -> impl Fn(&String) {
+        move |text_to_paste| {
+            let inserted = text_to_paste.as_str();
+            text_field_ptr.upgrade().for_each(|tf| { tf.borrow_mut().edit(inserted) })
+        }
+    }
+
+    fn char_typed_lambda(text_field_ptr:Weak<RefCell<TextFieldData>>) -> impl Fn(&Key,&KeyMask) {
+        move |key,mask| {
+            text_field_ptr.upgrade().for_each(|text_field| {
+                if let Key::Character(string) = key {
+                    let modifiers = &[Key::Control,Key::Alt,Key::Meta];
+                    if !modifiers.iter().any(|k| mask.has_key(k)) {
+                        text_field.borrow_mut().edit(string);
+                    }
+                }
+            })
+        }
+    }
+
+    fn initialize_actions_map
+    (actions:&mut KeyboardActions, text_field_ptr:Weak<RefCell<TextFieldData>>) {
+        use Key::*;
+        let mut setter = TextFieldActionsSetter{actions,text_field_ptr};
+        setter.set_navigation_action(&[ArrowLeft],    Step::Left);
+        setter.set_navigation_action(&[ArrowRight],   Step::Right);
+        setter.set_navigation_action(&[ArrowUp],      Step::Up);
+        setter.set_navigation_action(&[ArrowDown],    Step::Down);
+        setter.set_navigation_action(&[Home],         Step::LineBegin);
+        setter.set_navigation_action(&[End],          Step::LineEnd);
+        setter.set_navigation_action(&[Control,Home], Step::DocBegin);
+        setter.set_navigation_action(&[Control,End],  Step::DocEnd);
+        setter.set_action(&[Enter],     |t| t.edit("\n"));
+        setter.set_action(&[Delete],    |t| t.do_delete_operation(Step::Right));
+        setter.set_action(&[Backspace], |t| t.do_delete_operation(Step::Left));
```

Comment:
This is the initialization of keybindings? Is this a key manager or something like that? Anyway, intiialization of key bindings should be by passing a map of keybindoigs to some function. I mean, it sohuld be done by passing `HashMap<Key,Action>` to some kind of `initialize` function. We need to do it this way and prepare for proper shortcut manager (needed asap)